src/internaltypes.h

```diff
@@ -32,6 +39,128 @@ using EquivalenceMap = std::map<IndexStack, std::vector<IndexStack>>; /**< Type
 using NameList = std::vector<std::string>; /**< Type definition for list of names. */
 using StringStringMap = std::map<std::string, std::string>; /**< Type definition for map of string to string. */
 
+struct GeneratorEquationAst;
+using GeneratorEquationAstPtr = std::shared_ptr<GeneratorEquationAst>;
+using GeneratorEquationAstWeakPtr = std::weak_ptr<GeneratorEquationAst>;
+
+struct GeneratorEquationAst
```

Comment:
This isn't an internal type it is a full on declaration.

Reply:
I agree. The background to this is that I wanted/needed the `printAst()` method, so I added it to our `debug.[cpp|h]` file, which in turn implied that I had to add the `GeneratorEquationAst` structure to `internaltypes.h`. From there, I must confess that I haven't looked back, i.e. I haven't tried to "improve" that aspect of the PR. Will see later whether there is something I can do about it, however feel free to make suggestions, if you have any.

Reply:
This is fixed in commit 1075d7c6. @hsorby, is that fine with you? If so then feel free to resolve this conversation.

Reply:
I am resolving this since it has been done. If it is not suitable then another comment will anyway have to be made since this one is now considered outdated by GitHub.